.circleci/upload_test_coverage.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+# Script to upload code coverage to circleci
+
+project=$1
+if [[ $project == "" ]]
+then
+    echo "No project specified"
+    exit 1
+fi
+
+if [[ $project == "workers" ]]
+then
+    # the workers project uses it's own test_volume directory
+    test_volume="workers/test_volume"
+else
+    test_volume="test_volume"
+fi
+
+coverage_file="${test_volume}/coverage.xml"
+
+if [[ ! -f $coverage_file ]]
+then
+    echo "Coverage file wasn't found, were the tests run before?"
+    exit 0 # exit this script but don't fail the tests for this.
+fi
+
+output_file="${test_volume}/${project}_coverage.xml"
+
+# In the test coverage report, all file paths are absolute to each project
+# folder. We need to be relative to refinebio/ that's why we append the project
+# folder name to each file path in coverage.xml
+sed "s/filename=\"/filename=\"$project\//g" $coverage_file > $output_file
+
+curl -s https://codecov.io/bash | bash -s -- -f "$output_file" -Z -F $project
```

Comment:
Evaluating code that is pulled dynamically from the internet is a pretty big security flaw. Might I downloading that bash script, adding it to the repo, and then executing it from there. This way we know what we're executing and if codecov ever got hacked the hackers couldn't put a malicious script at that URL for us to run.

Reply:
This can also be installed as a [python package](https://github.com/codecov/codecov-python) do you think it's better if I do `pip install codecov` before this line?

Reply:
Works for me :+1: